core/camel-util/src/main/java/org/apache/camel/util/SensitiveUtils.java

```diff
@@ -95,6 +95,10 @@ private SensitiveUtils() {
     }
 
     public static boolean containsSensitive(String text) {
+        int lastPeriod = text.lastIndexOf('.');
+        if (lastPeriod >= 0) {
+            text = text.substring(lastPeriod + 1);
```

Comment:
Ah nice catch. Just a note that if the text ends with a dot, then you get a out of bounds exception.
Though it doesnt make sense to have keys that ends with a dot, but maybe an user make a typo

Reply:
No, `text.substring(text.length)` always returns an empty string.
I have added a test for a key ending with period in 8ade525 `assertFalse(SensitiveUtils.containsSensitive("foo.bar.accessKey."));` to make sure that it works as expected.